src/state.rs

```diff
@@ -468,9 +471,8 @@ impl ProgressDrawTarget {
             } => {
                 if draw_state.finished
                     || draw_state.force_draw
-                    || rate.is_none()
-                    || last_draw.is_none()
-                    || last_draw.unwrap().elapsed() > rate.unwrap()
+                    || rate == Duration::from_secs(0)
```

Comment:
I honestly don't like this use of a sentinel value. The benefit of saving 8 bytes doesn't really seem worth it to me.